### PR TITLE
ENH: Add Nan warnings for maximum, minimum on more dtypes

### PR DESCRIPTION
**Pull request by mattip**
Extends PR #11595 to `float16`, and complex types, which now set invalid float status when encountering a Nan in `np.maximum`, `np.minimum`

### CODE REVIEW

**Comment by charris** [comment]
What about `float{96, 128}` ?

**Comment by mattip** [comment]
`float{96,128}` already set invalid float status

File: numpy/core/src/umath/loops.c.src [hunk 1]
```diff
@@ -2222,9 +2222,13 @@ HALF_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED
 {
     /*  */
     BINARY_LOOP {
-        const npy_half in1 = *(npy_half *)ip1;
+        npy_half in1 = *(npy_half *)ip1;
         const npy_half in2 = *(npy_half *)ip2;
-        *((npy_half *)op1) = (@OP@(in1, in2) || npy_half_isnan(in1)) ? in1 : in2;
+        in1 = (@OP@(in1, in2) || npy_half_isnan(in1)) ? in1 : in2;
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply numpy/core/src/umath/loops.c.src]: Although I suppose it's consistent with what we already have for the others, so out of scope for this PR.

File: numpy/core/src/umath/loops.c.src [hunk 2]
```diff
@@ -2780,20 +2784,20 @@ NPY_NO_EXPORT void
 @TYPE@_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
     BINARY_LOOP {
-        const @ftype@ in1r = ((@ftype@ *)ip1)[0];
-        const @ftype@ in1i = ((@ftype@ *)ip1)[1];
+        @ftype@ in1r = ((@ftype@ *)ip1)[0];
```
Comment (charris) [inline numpy/core/src/umath/loops.c.src]: Hmm, now that I look at this, it is not quite the same as the sort ordering for complex numbers containing nans. Oh well.

**Comment by charris** [comment]
Thanks Matti.

**Pull request merged**